socketserver.h

```diff
@@ -262,21 +263,25 @@ class SocketServer
 
     bool ProcessIncomingConnectionsLoop(LEDBufferManager & bufferManager)
     {
+        // If an accept fails, we'll sleep this long in ms and try again
+        constexpr auto kSocketRetryDelay = 500;
+
         while (!interrupt_received)
         {
             if (0 >= _server_fd)
             {
                 printf("No _server_fd, returning.");
-                continue;
+                return false;
             }
 
-            int new_socket = 0;
-
             // Accept a new incoming connnection
+
+            int new_socket = 0;
             int addrlen = sizeof(_address);
             if ((new_socket = accept(_server_fd, (struct sockaddr *)&_address, (socklen_t*)&addrlen))<0)
             {
                 printf("Error accepting data!");
+                std::this_thread::sleep_for(std::chrono::milliseconds(kSocketRetryDelay));

```

Comment:
nit: you could declare `kSocketRetryDelay` as `std::chrono::milliseconds`, then you won't need this cast here